spec/predicates/deployment.md

```diff
@@ -0,0 +1,461 @@
+
+The decision to allow or deny a deployment request may happen in "real-time", i.e. the control plane may query an online authorization service at the time of the deployment. There are deployment scenario where querying an online service is not possible. For example, low-level stack like a VM bootloader may not have network access, on-prem deployments (e.g., using Anthos) may not allow network calls back to the cloud provider, etc.
+In situations where an online verification is technically possible, there are still devops reasons to perform an offline verification. An online authorization service requires low-latency / high-availability SLOs to avoid deployment outage. This is exacerbated in systems like Kubernetes where admission webhooks run for every pod deployed. Thus it is often desirable to "shift-left" and perform an authorization evaluation ahead of time before a deployment request reaches the control plane. The deployment attestation is the proof of authorization that the control plane may use to make its final decision, instead of querying an online service itself. Verification of the deployment attestation is simple, fast and may be performed entirely offline. Overall, this shift-left strategy provides the following advantages: less likely to cause production issues, better debugging UX for devs, less auditing and production noise for SREs and security teams.
+
+## Prerequisites
+
+This predicate depends on the [in-toto Attestation Framework](https://github.com/laurentsimon/attestation/blob/feat/deploy-att/spec/README.md).
+
+## Model
+
+This predicate is for the deployment stage of the software supply chain, where
+consumers want to bind an artifact to a deployment environment.
+An authorization service is called and generates a deployment attestation. The generated attestation can be embedded in the deployment request to deploy the artifact.
+The authorization service may be called explicitly in CI, or it may be called by the tooling without user involvement. Examples of tooling where this functionality could be implemented include [kubectl](https://kubernetes.io/docs/reference/kubectl/), [gcloud CLI](https://cloud.google.com/sdk/gcloud), [aws CLI](https://aws.amazon.com/cli/) or GitOps solutions like [ArgoCD](https://argo-cd.readthedocs.io/en/stable/).
+
+## Schema
+
+```jsonc
+{
+  // Standard attestation fields:
+  "_type": "https://in-toto.io/Statement/v1",
+  "subject": [{ ... }],
+
+  // Predicate:
+  "predicateType": "https://in-toto.io/attestation/deployment/v1",
+  "predicate": {
```

Comment:
Where is the bound environment described? I imagine the deployed artifact would be the `subject`, but I would have expected there to be some sort of required `environment` field, as `scope` is optional and seems to be specific about restrictions or controls placed upon the artifact within the environment.

Reply:
The environment is described via scopes. Scope is optional because you're allowed to express the fact that an artifact is allowed to be run in "any environment" if you wish to do that.